62_CAD/shaders/main_pipeline/fragment_shader.hlsl

```diff
@@ -408,11 +410,76 @@ float4 fragMain(PSInput input) : SV_TARGET
     float localAlpha = 0.0f;
     float3 textureColor = float3(0, 0, 0); // color sampled from a texture
 
-    // TODO[Przemek]: Disable All the object rendering paths if you want.
     ObjectType objType = input.getObjType();
     const uint32_t currentMainObjectIdx = input.getMainObjectIdx();
     const MainObject mainObj = mainObjects[currentMainObjectIdx];
-    
+
+    // TRIANGLE RENDERING
+    {
+        float3 v0 = input.getScreenSpaceVertexPos(0);
+        float3 v1 = input.getScreenSpaceVertexPos(1);
+        float3 v2 = input.getScreenSpaceVertexPos(2);
+
+        float2 start;
+        float2 end;
+        const float3 baryCoord = nbl::hlsl::spirv::BaryCoordKHR;
+
+        // TODO: figure out if branching can be reduced
+        // finding line start and end points by excluding vertex with the lowest barycentric coordinate value
+        if (baryCoord.x < baryCoord.y && baryCoord.x < baryCoord.z)
+        {
+            start = float2(v1.x, v1.y);
+            end = float2(v2.x, v2.y);
+        }
+        else if (baryCoord.y < baryCoord.x && baryCoord.y < baryCoord.z)
+        {
+            start = float2(v0.x, v0.y);
+            end = float2(v2.x, v2.y);
+        }
+        else if (baryCoord.z < baryCoord.x && baryCoord.z < baryCoord.y)
+        {
+            start = float2(v0.x, v0.y);
+            end = float2(v1.x, v1.y);
+        }
+
+        // long story short, in order for stipple patterns to be consistent:
+        // - point with lesser x coord should be starting point
+        // - if x coord of both points are equal then point with lesser y value should be starting point
+        if (end.x < start.x)
```

Comment:
do this checks to see whether to use `end` or `start` when constructing the line (before doing sdf of it in https://github.com/Devsh-Graphics-Programming/Nabla-Examples-and-Tests/pull/186/files#r2013523975)